src/lib/Core.jsx

```diff
@@ -313,48 +366,132 @@ function Core({
       {renderQuizResultQuestions()}
     </div>
   );
+
+  useEffect(() => {
+    let countdown;
+
+    if (timer && isRunning && timeRemaining > 0) {
+      countdown = setInterval(() => {
+        setTimeRemaining((prevTime) => prevTime - 1);
+      }, 1000);
+    }
+    return () => timer && clearInterval(countdown);
```

Comment:
what if the case is`timer = true, isRunning = false, timeRemaining = 0`? the countdown will be `undefined`.

Reply:
Addressed with validation on timer in Quiz component